tools/build_pak_files.py

```diff
@@ -16,13 +16,15 @@
 
     parser = argparse.ArgumentParser()
     parser.add_argument("--unreal_engine_dir", required=True)
-    parser.add_argument("--contents_dir", required=True)
+    parser.add_argument("--perforce_content_dir", required=True)
     parser.add_argument("--platforms", nargs="*", required=True)
     parser.add_argument("--output_dir", required=True)
     parser.add_argument("--scene_names")
     args = parser.parse_args()
     
     assert os.path.exists(args.unreal_engine_dir)
+    assert args.platforms in ["Win64", "Mac", "Linux"]
```

Comment:
Since `args.platform` can be a list, I don't think this assert statement doesn't work as intended.

Something like below should do it.
```python
all(platform in ["Win64", "Mac", "Linux"] for platform in args.platforms)
```

Reply:
Good catch. Upon closer inspection, cross-platform cooking doesn't work. I just to cook for Win64 on Mac and I got the following error.

```
LogTargetPlatformManager: Error: Invalid target platform specified (Win64NoEditor). Available = { Lumin, LuminClient, IOSClient, IOS, MacNoEditor, Mac, MacClient, MacServer, TVOSClient, TVOS, Android, Android_ASTC, Android_DXT, Android_ETC2, AndroidClient, Android_ASTCClient, Android_DXTClient, Android_ETC2Client, Android_Multi, Android_MultiClient }
```

Which makes sense. I wouldn't expect it to be possible to compile D3D shaders on Mac, or Metal shaders on Windows, etc. I'm going to change this script to only cook on the platform you're running the script on. I don't think this change is overly burdensome though, because you already need to be on Mac to build a Mac executable, so whatever machine is already doing that can also build the pak files.

Reply:
That makes sense! Yup, modifying the script to cook for the platform it's running on seems fine to me as well.